src/coreclr/jit/importer.cpp

```diff
@@ -4506,54 +4507,41 @@ GenTree* Compiler::impImportStaticFieldAccess(CORINFO_RESOLVED_TOKEN* pResolvedT
 
         default:
         {
-            // Do we need the address of a static field?
-            //
-            if (access & CORINFO_ACCESS_ADDRESS)
+// TODO-CQ: enable this optimization for 32 bit targets.
```

Comment:
No idea why this was disabled for 32bit previously (marking `static readonly` fields of references types as invariant) - I didn't change the behavior, will look separately at this.

Reply:
No, it is obviously an unnecessary pessimization.

(Whether it should be deleted in this change is a different question, likely the answer is no)